lib/rentlinx/client.rb

```diff
@@ -11,8 +11,39 @@ def initialize(username, password, api_url_prefix)
       @api_token ||= authenticate(username, password)
     end
 
+    def post(object)
+      case object
+      when Rentlinx::Property
+        post_property(object)
+      else
+        raise ProgrammerError, "Invalid object: #{object.class}"
```

Comment:
Would `TypeError` be more appropriate here? http://ruby-doc.org/core-2.2.0/TypeError.html